salt/_states/caasp_cmd.py

```diff
@@ -185,10 +188,33 @@ def run(name,
                                     **kwargs)
 
         if ret['result']:
-            return {'name': name,
-                    'changes': ret['changes'],
-                    'result': True,
-                    'comment': "Command executed succesfully after {0} retries. Last output: {1}".format(attempt + 1, ret['comment'])}
+            ret_success = {'name': name,
+                           'changes': ret['changes'],
+                           'result': True,
+                           'comment': "Command executed succesfully after {0} retries. Last output: {1}".format(attempt + 1, ret['comment'])}
+
+            # command run successful
+            if not retry_['until']:
+                return ret_success
+
+            # check if we are really done
+            retry_until_ret = __states__['cmd.run'](name=retry_['until'],
+                                                    cwd=cwd,
+                                                    runas=runas,
+                                                    shell=shell,
+                                                    env=env,
+                                                    umask=umask,
+                                                    output_loglevel=output_loglevel,
+                                                    quiet=quiet,
+                                                    timeout=timeout,
+                                                    ignore_timeout=ignore_timeout,
+                                                    use_vt=use_vt,
+                                                    **kwargs)
+            if retry_until_ret['result']:
+                return ret_success
+
+            # append the 'until' command output, so we can have some debugging info...
+            ret['comment'] = ret['comment'] + "(until: " + retry_until_ret['comment'] + ")"
```

Comment:
Is this meant for the last retry? Otherwise `ret` is being overriden on each loop pass.

Reply:
Yes, it is